Repro for `EE#154`: Sanboxed table should allow joins to it

nemanjaglumac opened the pull request:
### Status
READY

### What does this PR accomplish?
- Reproduces [EE_154](https://github.com/metabase/metabase-enterprise/issues/154)

### How to test this manually?
- build enterprise version of Metabase
- `yarn test-cypress-open --testFiles enterprise/frontend/test/metabase-enterprise`
- `enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js`
- Replace `it.skip()` with `it.only()` to test this in isolation
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

nemanjaglumac committed:
Add repro for `metabase-enterprise#154`

https://github.com/metabase/metabase-enterprise/issues/154

nemanjaglumac committed:
Update test from the sanboxed user perspective [ci skip]

flamber commented on enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:
```diff
@@ -275,6 +275,89 @@ describeWithToken("formatting > sandboxes", () => {
+
+      cy.findByText("Visualize").click();
+      cy.findByText("Count by User → ID");
+      cy.findAllByText("10");
```
Is this sandboxed user with attribute `user_id=3`? Otherwise I'm unsure what the `10` assertion is.

nemanjaglumac replied:
It is sandboxed user with the `user_id=1`. Accompanied `count` should be `11`. Great catch and an overlook on my side. I foolishly used the UI from unsandboxed (admin) results.

Thank you.

flamber reviewed (approved):
Not sure if my suggestions would work

nemanjaglumac commented on enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:
```diff
@@ -275,6 +275,89 @@ describeWithToken("formatting > sandboxes", () => {
+      });
+
+      // Go straight to orders table in custom questions
+      cy.visit("/question/new?database=1&table=2&mode=notebook");
```
This will be abstracted into a global helper function anyway. It will use variables instead of hard coded values.

flamber replied:
My point is that it's much easier to read variables instead of guessing what table ID 2 is supposed to be.

nemanjaglumac replied:
Absolutely agreed!
I'm just saying that we're probably a long overdue abstracting this and making it global. It didn't make sense that I change it here in one place only when I plan to use helper function instead.

But you have the point and it will be addressed.

nemanjaglumac commented on enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js:
```diff
@@ -275,6 +275,89 @@ describeWithToken("formatting > sandboxes", () => {
+      cy.log(
+        "**-- Original issue reported failure to find 'User' group / foreign key--**",
+      );
+      cy.get(".ReactVirtualized__Grid").scrollTo("bottom"); // Cypress scroll sometimes doesn't work; We should disable virtualization for Popover
```
It seems to be working locally. Excellent idea! 🚀

nemanjaglumac committed:
Improve test [ci skip]

- Fix test assertion
- Use UI trick instead of scrolling to get around virtualization
